integrate.js

```diff
@@ -47,17 +47,24 @@ var PlayerAction = Nuvola.PlayerAction;
 // Create new WebApp prototype
 var WebApp = Nuvola.$WebApp();
 
+// Translations
+var C_ = Nuvola.Translate.pgettext;
+
+var ACTION_LIKE = "like";
+
 // Initialization routines
 WebApp._onInitWebWorker = function(emitter)
 {
     Nuvola.WebApp._onInitWebWorker.call(this, emitter);
-    
-    
+
     var state = document.readyState;
     if (state === "interactive" || state === "complete")
         this._onPageReady();
     else
         document.addEventListener("DOMContentLoaded", this._onPageReady.bind(this));
+
+    Nuvola.actions.addAction("like", "win", ACTION_LIKE, C_("Action", "Like"),
+        null, null, null, null);
```

Comment:
Isn't the Like action a toggle action (on/off)? I mean do you press the button once to like the track and then again to unlike it? For example, the thumbs up action in Google Play and in [Custom Actions](https://tiliado.github.io/nuvolaplayer/development/apps/custom-actions.html) is a toggle action.

Reply:
Yes, like it's toggle action. Do you mean that there should be separate actions for like and unlike?

Reply:
No, a single action but defined as a toggle action. The key is the last parameter of [Actions.addAction](https://tiliado.github.io/nuvolaplayer/development/apps/api_reference.html#Nuvola.Actions.addAction). It's `null` for simple action and `true/false` for toggle actions:
```js
Nuvola.actions.addAction(
    "playback", "win", ACTION_LIKE, C_("Action", "Like"),
    null, null, null, false);
```

Then you need update the current state in your `WebApp.update()` function with [Nuvola.actions.updateState](https://tiliado.github.io/nuvolaplayer/development/apps/api_reference.html#Nuvola.Actions.updateState).

```js
Nuvola.actions.updateState(ACTION_LIKE, true); // The song is liked
Nuvola.actions.updateState(ACTION_LIKE, false); // The song is not liked
```

Then test them in the developer sidebar. Simple actions are shown as buttons, toggle actions as checkboxes.

![custom_actions_enabled](https://user-images.githubusercontent.com/853706/33796517-1dea7ef8-dcf6-11e7-962d-2f0582520498.png)